contracts/src/access/test/ShieldedAccessControl.test.ts

```diff
@@ -0,0 +1,991 @@
+            ),
+          ).not.toEqual(ADMIN_SECRET_NONCE);
+        }
+
+        if (isValidIndex) {
+          // Check index matches
+          const [, witnessCalculatedIndex] =
+            shieldedAccessControl.witnesses.wit_getRoleIndex(
+              shieldedAccessControl.getWitnessContext(),
+              DEFAULT_ADMIN_ROLE,
+              Z_ADMIN
+            );
+          expect(witnessCalculatedIndex).toBe(INIT_COUNTER);
+        } else {
+          // Check index does not match
+          shieldedAccessControl.overrideWitness(
+            'wit_getRoleIndex',
+            RETURN_BAD_INDEX,
+          );
+          const [, witnessCalculatedIndex] =
+            shieldedAccessControl.witnesses.wit_getRoleIndex(
+              shieldedAccessControl.getWitnessContext(),
+              DEFAULT_ADMIN_ROLE,
+              Z_ADMIN
+            );
+          expect(witnessCalculatedIndex).not.toBe(INIT_COUNTER);
+        }
+
+        if (isValidPath) {
+          // Check path matches
+          const truePath = shieldedAccessControl
+            .getPublicState()
+            .ShieldedAccessControl__operatorRoles.findPathForLeaf(
+              EXP_DEFAULT_ADMIN_COMMITMENT,
+            );
+          const [, witnessCalculatedPath] =
+            shieldedAccessControl.witnesses.wit_getRoleCommitmentPath(
+              shieldedAccessControl.getWitnessContext(),
+              EXP_DEFAULT_ADMIN_COMMITMENT,
+            );
+          expect(witnessCalculatedPath).toEqual(truePath);
+        } else {
+          // Check path does not match
+          const truePath = shieldedAccessControl
+            .getPublicState()
+            .ShieldedAccessControl__operatorRoles.findPathForLeaf(
+              EXP_DEFAULT_ADMIN_COMMITMENT,
+            );
+          shieldedAccessControl.overrideWitness(
+            'wit_getRoleCommitmentPath',
+            RETURN_BAD_PATH,
+          );
+          const [, witnessCalculatedPath] =
+            shieldedAccessControl.witnesses.wit_getRoleIndex(
+              shieldedAccessControl.getWitnessContext(),
+              DEFAULT_ADMIN_ROLE,
+              Z_ADMIN
+            );
+          expect(witnessCalculatedPath).not.toEqual(truePath);
+        }
+
+        // Test false case circuit
+        const role = (
+          shieldedAccessControl['hasRole'] as (...args: unknown[]) => Role
+        )(...args);
+        expect(role.roleCommitment).not.toEqual(EXP_DEFAULT_ADMIN_COMMITMENT);
+      },
+    );
+  });
+
+  describe('assertOnlyRole', () => {
+    beforeEach(() => {
+      shieldedAccessControl._grantRole(DEFAULT_ADMIN_ROLE, Z_ADMIN);
+      shieldedAccessControl.callerCtx.setCaller(ADMIN);
+    });
+
+    it('should not fail when authorized caller has correct nonce, index, and path', () => {
+      // Check nonce is correct
+      expect(
+        shieldedAccessControl.privateState.getCurrentSecretNonce(
+          DEFAULT_ADMIN_ROLE,
+        ),
+      ).toBe(ADMIN_SECRET_NONCE);
+
```

Comment:
I think there's something wrong because this passes:

```
shieldedAccessControl.callerCtx.setCaller('maximus decimus meridius');
shieldedAccessControl.assertOnlyRole(new Uint8Array(32).fill(123456789));
```

Reply:
Hmmm, I'm seeing it fail due to Compact address formatting issues unrelated to the contract itself

Reply:
My bad, use this:

```
const [maximus, z_maximus] = utils.generateEitherPubKeyPair('maximus decimus meridius');

shieldedAccessControl.callerCtx.setCaller(maximus);
shieldedAccessControl.assertOnlyRole(new Uint8Array(32).fill(123456789));
```

lmk if I'm mistaken